File: Server/FIHS/Models/PestModels/Pest.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace FIHS.Models.PestModels
{
    public class Pest
    {
        public int Id { get; set; }
        [MinLength(2),MaxLength(50)]
        public string Species { get; set; }
        [MinLength(2),MaxLength(50)]
        public string Name { get; set; }
        [MinLength(2),MaxLength(128)]
        public string ScientificName { get; set; }
        public string ImageUrl { get; set; }
        [MinLength(2), MaxLength(128)]
        public string DamageSymptoms { get; set; }
        [MinLength(2), MaxLength(128)]
        public string ControlMethods { get; set; }
        [MinLength(2), MaxLength(128)]
        public string Description { get; set; }
        [MinLength(2), MaxLength(128)]
        public string LifeCycle { get; set; }
        [MinLength(2), MaxLength(50)]
        public string GeographicDistribution { get; set; }
        [MinLength(2), MaxLength(128)]
        public string  Reproduction { get; set; }
        [JsonIgnore,ValidateNever,NotMapped]
        public string Message { get; set; }
        [ValidateNever, JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual IEnumerable<PlantsPests> Plants { get; set; }
        [ValidateNever]
        public virtual IEnumerable<PestsPesticides> Pesticides { get; set; }

    }
}

File: Client/FIHS/models/Pest.ts
import * as yup from 'yup'

export const PestSchema = yup.object({
    species: yup.string().required('برجاء إدخال الفصيلة'),
    name: yup.string().required('برجاء إدخال الاسم'),
    scientificName: yup.string().required('برجاء إدخال الاسم العلمي'),
    image: yup.mixed().required('برجاء إرفاق صورة'),
    description: yup.string().required('برجاء إدخال الوصف'),
    damageSymptoms: yup.string().required('برجاء إدخال الاعراض'),
    controlMethods: yup.string().required('برجاء إدخال طرق الوقاية'),
    lifeCycle: yup.string().required('برجاء إدخال دورة الحياة'),
    geographicDistribution: yup.string().required('برجاء إدخال اماكن الانتشار'),
    reproduction: yup.string().required('برجاء إدخال طرق التكاثر'),
})
export type Pest = yup.InferType<typeof PestSchema>